### PR TITLE
Fix nil-pointer dereference on windows context

### PR DESCRIPTION
**Pull request by thedadams**
Issue: https://github.com/rancher/rancher/issues/37628

Passing certContext to the deferred function ensures that the value will
be taken at that point. The issue is that it is nil when the deferred
function is called.

This change will capture the variable so its real value is passed

### CODE REVIEW

File: pkg/prober/prober_windows.go [hunk 1]
```diff
@@ -43,9 +43,11 @@ func GetSystemCertPool(probeName string) (*x509.CertPool, error) {
 	var certContext *syscall.CertContext
 
 	// ref for why flags value is 0: https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certclosestore
-	defer func(store syscall.Handle, flags uint32) {
-		_ = syscall.CertCloseStore(store, flags)
-	}(certContext.Store, 0)
+	defer func() {
```
Comment (rosskirkpat) [inline pkg/prober/prober_windows.go:46]: The defer needs to be moved below the for loop on L94. I missed this on my initial PR, sorry!

**Review by rosskirkpat — CHANGES_REQUESTED** [review]
move the original defer to L95 https://github.com/rancher/system-agent/blob/46fbba3a2040e7f2f4ef67802ee867dfe1cab9e8/pkg/prober/prober_windows.go#L95, after the for loop

File: pkg/prober/prober_windows.go (hunk 1 above)
Comment (thedadams) [reply pkg/prober/prober_windows.go:46]: Why does it need to be moved?

Comment (rosskirkpat) [reply pkg/prober/prober_windows.go:46]: in addition to moving the defer, you would also do your original nil check after the for loop

**Review by luthermonson — APPROVED** [review]

**Comment by rosskirkpat** [comment]
Were you able to validate this fixes the issue? You would need to pin this SA commit in a fork of wins and then compile wins and push the binary to a windows test machine.

**Comment by luthermonson** [comment]
tested and the node came up

**Review by rosskirkpat — APPROVED** [review]
lgtm, thank you for testing!

**Pull request merged**